### PR TITLE
fix: Use starkbank/ecdsa 2.1.0

### LINKED ISSUE #1108: Deprecated Property Warnings in EventWebhook Signature Verification with PHP 8.2
### Issue Summary
The issue involves using the convertPublicKeyToECDSA and verifySignature functions in the SendGrid PHP client library. When executing these functions, deprecated warnings are logged.

### Steps to Reproduce
1. Call the convertPublicKeyToECDSA function with the SendGrid webhook public key.
2. Use the output of the first function as input to the verifySignature function along with the request content and headers.

### Code Snippet
```php
$eventWebhook = new EventWebhook();
$ecPublicKey = $eventWebhook->convertPublicKeyToECDSA($_ENV['SENDGRID_WEBHOOK_PUBLIC_KEY']);
return $eventWebhook->verifySignature(
            $ecPublicKey,
            $request->getContent(),
            $request->headers->get(EventWebhookHeader::SIGNATURE),
            $request->headers->get(EventWebhookHeader::TIMESTAMP)
        );
```

### Exception/Log
```
Deprecated: Creation of dynamic property EllipticCurve\PublicKey::$pem is deprecated
Deprecated: Creation of dynamic property EllipticCurve\Signature::$der is deprecated
```

### Technical details:
* sendgrid-php version: 8.1.1
* php version: 8.2.19

### PR DESCRIPTION
**Pull request by simoheinonen**
# Fixes #

This PR bumps the `starkbank/ecdsa` version to `^2.1.0` so we get rid of deprecations.

- fixes https://github.com/sendgrid/sendgrid-php/issues/1108

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-php/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://support.sendgrid.com).

### CODE REVIEW

File: test/unit/EventWebhookTest.php [hunk 1]
```diff
@@ -23,6 +24,8 @@ public static function setUpBeforeClass(): void
         IW4mdBgQ/7dAfSmpqIM8kF9mN1flpVKS3GRqe62gw+2fNNRaINXvVpiglSI8eNEc6wEA3F+g==';
         self::$SIGNATURE = 'MEUCIGHQVtGj+Y3LkG9fLcxf3qfI10QysgDWmMOVmxG0u6ZUAiE
         AyBiXDWzM+uOe5W0JuG+luQAbPIqHh89M15TluLtEZtM=';
+        self::$BAD_SIGNATURE = 'BADSIGNATURE+Y3LkG9fLcxf3qfI10QysgDWmMOVmxG0u6ZUAiE
+        AyBiXDWzM+uOe5W0JuG+luQAbPIqHh89M15TluLtEZtM=';
```
Comment (simoheinonen) [inline test/unit/EventWebhookTest.php:28]: `Exception: missing bytes in DER parse` is thrown if this bad signature isn't in the right format

**Comment by vincent-lu** [comment]
Can someone please take a look at this PR? Current `sendgrid-php` is not working with the latest version of `starkbank/ecdsa-php` library.

See: https://github.com/starkbank/ecdsa-php/issues/23

**Comment by 54853315** [comment]
No reviews? Hmm, I think you should email [support](https://support.sendgrid.com/hc/en-us?_gl=1*3udvtx*_gcl_au*MTQzMDUyNDI3OS4xNzMyMDY1NzE3*_ga*MjA1NjIyMTY4OS4xNzMyMDY1NzE3*_ga_8W5LR442LD*MTczMjA2NTcxNy4xLjEuMTczMjA3MDUxMC4wLjAuMA).

**Comment by tiwarishubham635** [comment]
Let me check

**Comment by tiwarishubham635** [comment]
@simoheinonen some tests seem to fail. Can you check once?

**Comment by tiwarishubham635** [comment]
Looks good now. Merging

**Review by tiwarishubham635 — APPROVED** [review]

**Comment by vincent-lu** [comment]
Thanks guys!

**Comment by tectiv3** [comment]
This should've been at least a minor release change. 

This package now suddenly requires `gmp` extension and that change definitely came unexpected.

**Comment by vincent-lu** [comment]
Hi @tiwarishubham635 is there any reason why `gmp` extension is needed here? I checked the `sendgrid-php` source it doesn't seem to call any gmp related functions?

**Pull request merged**